File: Go/binarytree.go
// Binary tree implementation in Go

package main

type Node struct {
	value int
	left  *Node
	right *Node
}

func (n *Node) insert(value int) {
	// your code here
}
func (n *Node) printTree() {
	// your code here
}

func main() {
	root := Node{value: 100}
	root.insert(50)
	root.insert(200)
	root.insert(25)
	root.printTree()
}

File: Rust/binarytree.rs
// Binary Tree in Rust

fn main() {
    let tree = Node::new(100);
    let tree = tree.insert(200);
    let tree = tree.insert(50);
    let tree = tree.insert(25);
    tree.printtree();
}

struct Node {
    value: i32,
    left: Option<Box<Node>>,
    right: Option<Box<Node>>,
}

impl Node {
    fn new(value: i32) -> Node {
        Node {
            value,
            left: None,
            right: None,
        }
    }

    fn insert(&self, value: i32) -> Node {
        // your code goes here
    }
    fn print_tree(&self) {
        // your code goes here
    }
}
